pkg/sql/schemachanger/scbuild/internal/scbuildstmt/helpers.go

```diff
@@ -493,29 +533,47 @@ func makeIndexColumnSpec(ic *scpb.IndexColumn) indexColumnSpec {
 	}
 }
 
-// makeSwapPrimaryIndexSpec constructs a primaryIndexSpec and an accompanying
-// tempIndexSpec to swap out an existing primary index with.
-func makeSwapPrimaryIndexSpec(
-	b BuildCtx, out primaryIndexSpec, inColumns []indexColumnSpec,
-) (in primaryIndexSpec, temp tempIndexSpec) {
+// makeSwapIndexSpec constructs a pair of indexSpec for the new index and the
+// accompanying temporary index to swap out an existing index with.
+func makeSwapIndexSpec(
```

Comment:
this code implicitly assumes that the spec is either primary or secondary, but I don't see code anywhere to assert that that's true. I'm fine with having this code assume `out` is well-formed, but something above it needs to assert that

Reply:
Fixed.